runner/proxy_stream.go

```diff
@@ -42,13 +61,36 @@ func (stream *ProxyStream) RemoveSubscriber(id int32) {
 	defer stream.mu.Unlock()
 
 	if _, ok := stream.subscribers[id]; ok {
-		close(stream.subscribers[id].reloadCh)
+		close(stream.subscribers[id].msgCh)
 		delete(stream.subscribers, id)
 	}
 }
 
 func (stream *ProxyStream) Reload() {
 	for _, sub := range stream.subscribers {
-		sub.reloadCh <- struct{}{}
+		sub.msgCh <- StreamMessage{
+			Type: StreamMessageReload,
+			Data: nil,
+		}
 	}
 }
+
+func (stream *ProxyStream) BuildFailed(err BuildFailedMsg) {
+	for _, sub := range stream.subscribers {
+		sub.msgCh <- StreamMessage{
+			Type: StreamMessageBuildFailed,
+			Data: err,
+		}
+	}
+}
+
+func (m StreamMessage) AsSSE() string {
+	s := "event: " + string(m.Type) + "\n"
+	s += "data: " + stringify(m.Data) + "\n"
+	return s + "\n"
+}
+
+func stringify(v any) string {
+	b, _ := json.Marshal(v)
```

Comment:
if err != nil {
        return fmt.Sprintf("{\"error\":\"Failed to marshal message: %s\"}", err)
    }